modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/exception/DistributionZoneBindTableException.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.distributionzones.exception;
+
+import static org.apache.ignite.lang.ErrorGroups.DistributionZones.ZONE_BIND_TABLE_ERR;
+
+import java.util.UUID;
+import org.apache.ignite.lang.IgniteInternalException;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Exception is thrown when the distribution zone cannot be dropped because there is a table bound to the distribution zone.
+ */
+public class DistributionZoneBindTableException extends IgniteInternalException {
+    /**
+     * The constructor.
+     *
+     * @param zoneName Zone name.
+     */
+    public DistributionZoneBindTableException(String zoneName) {
+        this(zoneName, null);
+    }
+
+    /**
+     * The constructor.
+     *
+     * @param zoneName Zone name.
+     * @param cause Optional nested exception (can be {@code null}).
+     */
+    public DistributionZoneBindTableException(String zoneName, @Nullable Throwable cause) {
+        super(ZONE_BIND_TABLE_ERR, "Distribution zone is assigned to the table [zoneName=" + zoneName + ']', cause);
```

Comment:
The exception throws after first bound table is found, not full list of bound tables. So I think it is not so useful to know only one table name.

Reply:
I don't insist, but I would also suggest listing all the related tables in the exception message. It seems that this may make life easier for someone in the future.

Reply:
OK. I added table name.